2000-08-18  Joseph S. Myers  <jsm28@cam.ac.uk>

	* gcc.dg/format-va-1.c: New test.

--- gcc/testsuite/gcc.dg/format-va-1.c	Fri Sep 11 11:31:59 1998
+++ gcc/testsuite/gcc.dg/format-va-1.c	Fri Aug 18 01:31:57 2000
@@ -0,0 +1,13 @@
+/* Test for strange warning in format checking.  */
+/* Origin: Joseph Myers <jsm28@cam.ac.uk> */
+/* { dg-do compile } */
+/* { dg-options "-Wformat" } */
+
+extern int printf (const char *, ...);
+
+void
+foo (void *p)
+{
+  printf ("%d", p); /* { dg-bogus "va_list" "wrong type in format warning" } */
+  /* { dg-warning "format" "format error" { target *-*-* } 11 } */
+}
